packages/headless/src/api/platform-client.ts

```diff
@@ -40,8 +40,14 @@ export const NoopPreprocessRequestMiddleware: PreprocessRequestMiddleware = (
   request
 ) => request;
 
+export enum PlatformClientErrorResponse {
+  Disconnected,
+}
+
 export class PlatformClient {
-  static async call(options: PlatformClientCallOptions): Promise<Response> {
+  static async call(
+    options: PlatformClientCallOptions
+  ): Promise<Response | PlatformClientErrorResponse> {
```

Comment:
I had to change the return type of `call`, because when there's no internet connection then `fetch` throws an error instead of returning a `Response` with a status code. Having an enum like this (or even a string type alias) forces every caller to handle the possible lack of a response.

Reply:
A problem with multiple return types is all functions using the `call` method need to be adapted to handle each return type. All the methods in search-api and case-assist clients needed to be adapted following this change, which makes the code-base harder to extend and adapt in the future.

Would it be possible to catch the error upstream instead, and handle it there similar to other errors?

Reply:
I can fake a HTTP response